sympy/printing/c.py

```diff
@@ -387,6 +395,8 @@ def _print_MatrixElement(self, expr):
 
     def _print_Symbol(self, expr):
         name = super()._print_Symbol(expr)
+        # if name is an indexed, e.g. a[0], only check the variable name
+        _is_valid_c_variable_name(name.split('[')[0])
```

Comment:
I suspect we will need to guard this with a printer option (opt-in) for back-comp reasons.